src/compiler/transformers/ts.ts

```diff
@@ -3364,7 +3368,7 @@ namespace ts {
                         ? declarationNameToString(originalNode.name)
                         : getTextOfNode(originalNode.argumentExpression);

```

Comment:
Move the function call to this line so that the common case (where a user has written a property access and isn't trying to sabotage the emitter 😄) doesn't have to run through a regex